### PR TITLE
[rtsp_to_hls] Update

### PR DESCRIPTION
**Pull request by varsill**
This PR:
* updates `rtsp_to_hls` demo to Core v1.0
* removes unused dependency to FFmpeg
* adds Elixir implementation of a simple RTSP server to be run as `server.exs` script
* removes application setup and instead creates `rtsp_to_hls.exs` script
* updates README

### CODE REVIEW

File: rtsp_to_hls/README.md [hunk 1]
```diff
@@ -44,137 +29,50 @@ git clone https://github.com/membraneframework/membrane_demo
...
-mix run --no-halt
-```
-
-After a moment the pipeline will start generating HLS output files. To watch the stream we need to serve those files, e.g. by using python HTTP server:
+After a moment the pipeline will start generating HLS output files (by default in the `hls_output` directory).
+To watch the stream we need to serve those files, e.g. by using python HTTP server:
 
 ```shell
+cd hls_output
 python3 -m http.server 8000
```
Comment (mat-hek) [inline rtsp_to_hls/README.md:73]: You can use erlang's built-in server, like in the camera_to_hls demo

File: rtsp_to_hls/lib/connection_manager.ex [hunk 2]
```diff
@@ -217,7 +218,7 @@ defmodule Membrane.Demo.RtspToHls.ConnectionManager do
     end
   end
 
-  defp play(%ConnectionStatus{rtsp_session: rtsp_session, pipeline: pipeline}) do
+  defp play(%ConnectionStatus{rtsp_session: rtsp_session, pipeline: _pipeline}) do
```
Comment (mat-hek) [inline rtsp_to_hls/lib/connection_manager.ex:221]:
```suggestion
  defp play(%ConnectionStatus{rtsp_session: rtsp_session}) do
```

File: rtsp_to_hls/lib/pipeline.ex [hunk 3]
```diff
@@ -115,8 +127,20 @@ defmodule Membrane.Demo.RtspToHls.Pipeline do
...
   def handle_child_notification(notification, element, _ctx, state) do
-    Logger.warn("Unknown notification: #{inspect(notification)}, el: #{inspect(element)}")
+    Logger.warning("Unknown notification: #{inspect(notification)}, el: #{inspect(element)}")
 
     {[], state}
   end
+
+  @impl true
+  def handle_element_end_of_stream(Pad.ref(:input), _pad, _context, state) do
+    IO.inspect("terminating")
```
Comment (mat-hek) [inline rtsp_to_hls/lib/pipeline.ex:137]: leftover

File: rtsp_to_hls/lib/pipeline.ex [hunk 4]
```diff
@@ -115,8 +127,20 @@ defmodule Membrane.Demo.RtspToHls.Pipeline do
...
+    {[terminate: :normal], state}
+  end
+
+  @impl true
+  def handle_element_end_of_stream(_child, _pad, _context, state) do
+    IO.inspect("terminating2")
```
Comment (mat-hek) [inline rtsp_to_hls/lib/pipeline.ex:143]: leftover

File: rtsp_to_hls/lib/pipeline.ex [hunk 5]
```diff
@@ -115,8 +127,20 @@ defmodule Membrane.Demo.RtspToHls.Pipeline do
...
   @impl true
   def handle_child_notification(notification, element, _ctx, state) do
-    Logger.warn("Unknown notification: #{inspect(notification)}, el: #{inspect(element)}")
+    Logger.warning("Unknown notification: #{inspect(notification)}, el: #{inspect(element)}")
 
     {[], state}
   end
+
+  @impl true
+  def handle_element_end_of_stream(Pad.ref(:input), _pad, _context, state) do
```
Comment (mat-hek) [inline rtsp_to_hls/lib/pipeline.ex:136]: This won't work, see https://hexdocs.pm/membrane_core/Membrane.Pipeline.html#c:handle_element_end_of_stream/4

File: rtsp_to_hls/rtsp_to_hls.exs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+alias Membrane.Demo.RtspToHls.Pipeline
```
Comment (mat-hek) [inline rtsp_to_hls/rtsp_to_hls.exs:1]:
```suggestion
alias Membrane.Demo.RtspToHls
```

File: rtsp_to_hls/server.exs [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        exit(:normal)
+
+      {:error, reason} ->
+        raise("Error when getting request: #{inspect(reason)}")
+    end
+  end
+
+  defp generate_response(request_type, state) do
+    case request_type do
+    :describe ->
```
Comment (mat-hek) [inline rtsp_to_hls/server.exs:199]: seems like this is not formatted

File: rtsp_to_hls/README.md [hunk 8]
```diff
@@ -44,137 +29,50 @@ git clone https://github.com/membraneframework/membrane_demo
...
-You can configure the parameters for the converter in the `Application` module:
-
-##### lib/application.ex
+##### rtsp_to_hls.exs
 
 ```elixir
-@rtsp_stream_url "rtsp://rtsp.membrane.work:554/testsrc.264"
+@rtsp_stream_url "rtsp://localhost:8554/livestream"
 @output_path "hls_output"
 @rtp_port 20000
```
Comment (mat-hek) [inline rtsp_to_hls/README.md:65]:
```suggestion
rtsp_stream_url = "rtsp://localhost:8554/livestream"
output_path = "hls_output"
rtp_port = 20000
```

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: rtsp_to_hls/lib/pipeline.ex (hunk 4 above)
Comment (varsill) [reply rtsp_to_hls/lib/pipeline.ex:143]: done

File: rtsp_to_hls/lib/pipeline.ex (hunk 3 above)
Comment (varsill) [reply rtsp_to_hls/lib/pipeline.ex:137]: done

File: rtsp_to_hls/rtsp_to_hls.exs (hunk 6 above)
Comment (varsill) [reply rtsp_to_hls/rtsp_to_hls.exs:1]: done

File: rtsp_to_hls/server.exs (hunk 7 above)
Comment (varsill) [reply rtsp_to_hls/server.exs:199]: indeed, I have added these two script files to the formatter config

File: rtsp_to_hls/README.md (hunk 8 above)
Comment (varsill) [reply rtsp_to_hls/README.md:65]: done

File: rtsp_to_hls/lib/pipeline.ex (hunk 5 above)
Comment (varsill) [reply rtsp_to_hls/lib/pipeline.ex:136]: As discusses, I have removed the callback implementation

File: rtsp_to_hls/.formatter.exs [hunk 9]
```diff
@@ -1,4 +1,9 @@
 # Used by "mix format"
 [
-  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
+  inputs: [
+    "{mix,.formatter}.exs",
+    "{config,lib,test}/**/*.{ex,exs}",
+    "server.exs",
+    "rtsp_to_hls.exs"
```
Comment (mat-hek) [inline rtsp_to_hls/.formatter.exs:7]:
```suggestion
    "{config,lib,test}/**/*.{ex,exs}",
    "*.exs"
```

**Pull request merged**